src/dbnode/server/server.go

```diff
@@ -1165,6 +1167,81 @@ func kvWatchEncodersPerBlockLimit(
 	}()
 }
 
+func kvWatchQueryLimit(
+	store kv.Store,
+	logger *zap.Logger,
+	limits limits.QueryLimits,
+) {
+	value, err := store.Get(kvconfig.QueryLimits)
+	if err == nil {
+		protoValue := &kvpb.QueryLimits{}
+		err = value.Unmarshal(protoValue)
+		if err == nil && protoValue != nil {
+			updateQueryLimits(logger, limits, protoValue)
+		}
+	} else if !errors.Is(err, kv.ErrNotFound) {
+		logger.Warn("error resolving query limit", zap.Error(err))
+	}
+
+	watch, err := store.Watch(kvconfig.QueryLimits)
+	if err != nil {
+		logger.Error("could not watch query limit", zap.Error(err))
+		return
+	}
+
+	go func() {
+		protoValue := &kvpb.QueryLimits{}
+		for range watch.C() {
+			if newValue := watch.Get(); newValue != nil {
```

Comment:
nit: flip condition to reduce nesting

Reply:
I think we want the newly set `newValue` within though